### PR TITLE
Fix missing lamp fire in Lazarus' chambers

### PR DESCRIPTION
**Pull request by AJenbo**
Also cleanup object update code

image

### CODE REVIEW

**Comment by FitzRoyX** [comment]
sarcoffagus -> sarcophagus

File: Source/objects.cpp [hunk 1]
```diff
@@ -1511,37 +1511,37 @@ void UpdateCircle(int i)
...
-	Objects[i]._oSelFlag = 2;
-	Objects[i]._oVar4 = dok ? 1 : 2;
-	Objects[i]._oMissFlag = true;
+	door._oSelFlag = 2;
+	door._oVar4 = dok ? 1 : 2;
+	door._oMissFlag = true;
 }
 
-void UpdateSarcoffagus(int i)
+void UpdateSarcoffagus(Object &sarcoffagus)
```
Comment (AJenbo) [inline Source/objects.cpp:1541]:
```suggestion
void UpdateSarcophagus(Object &sarcophagus)
```

File: Source/objects.cpp [hunk 2]
```diff
@@ -1511,37 +1511,37 @@ void UpdateCircle(int i)
...
 {
-	if (Objects[i]._oAnimFrame == Objects[i]._oAnimLen)
-		Objects[i]._oAnimFlag = 0;
+	if (sarcoffagus._oAnimFrame == sarcoffagus._oAnimLen)
+		sarcoffagus._oAnimFlag = 0;
```
Comment (AJenbo) [inline Source/objects.cpp:1544]:
```suggestion
	if (sarcophagus._oAnimFrame == sarcophagus._oAnimLen)
		sarcophagus._oAnimFlag = 0;
```

File: Source/objects.cpp [hunk 3]
```diff
@@ -5065,49 +5065,49 @@ void ProcessObjects()
...
 		case OBJ_TORCHR:
 		case OBJ_TORCHL2:
 		case OBJ_TORCHR2:
-			UpdateObjectLight(oi, 8);
+			UpdateObjectLight(object, 8);
 			break;
 		case OBJ_SARC:
 		case OBJ_L5SARC:
-			UpdateSarcoffagus(oi);
+			UpdateSarcoffagus(object);
```
Comment (AJenbo) [inline Source/objects.cpp:5078]:
```suggestion
			UpdateSarcophagus(object);
```

**Review by obligaron — COMMENTED** [review]
lgtm 🙂 
Didn't used approve, cause this would trigger the auto-merge and I wasn't sure if the last commit (Correct typo) should be combined with first commit before merging.

**Comment by AJenbo** [comment]
Thanks, since the typo goes back before this PR I think I will keep it separate.

**Pull request merged**